soda_machine.rb

```diff
@@ -7,12 +7,28 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    counter = 0
+    @sodas.each do |sodas|
+      counter += 1
+    end
+      counter
   end
 
   def find_soda(soda_brand)
+    found_soda = []
+    @sodas.select do |soda|
+      if soda.brand == soda_brand
+          found_soda << soda
+      else
+        nil
+      end
```

Comment:
What if the first one doesn't match but a later one does?  (Unfortunately our tests didn't test that case.)